govcd/disk.go

```diff
@@ -227,13 +227,20 @@ func (disk *Disk) Delete() (Task, error) {
 func (disk *Disk) Refresh() error {
 	util.Logger.Printf("[TRACE] Disk refresh, HREF: %s\n", disk.Disk.HREF)
 
-	fetchedDisk, err := FindDiskByHREF(disk.client, disk.Disk.HREF)
+	if *disk == (Disk{}) {
+		return fmt.Errorf("cannot refresh, Object is empty")
+	}
+
+	unmarshalledDisk := &types.Disk{}
+
+	_, err := disk.client.ExecuteRequest(disk.Disk.HREF, http.MethodGet,
+		"", "error refreshing independent disk: %s", nil, unmarshalledDisk)
```

Comment:
Suggestion: use `GetDiskByHref` instead of the low-level `ExecuteRequest`, so all disk HTTP request logic goes through this single function.

Reply:
GetDiskByHref requires VDC, which we don't have here

Reply:
Don't we know VDC from Disk?

Reply:
If I would figure you vdc name, then I need to make request/search to fetch VDC object and then I can request that function

Reply:
Yeah, sounds introducing a hop - an additional call where we can live without it.